src/lib/provable/field.ts

```diff
@@ -798,12 +786,14 @@ class Field {
     try {
       if (this.isConstant()) {
         let x = this.toBigInt();
-        if (x !== 0n && x !== 1n) {
-          throw Error(`Field.assertBool(): expected ${x} to be 0 or 1`);
-        }
-        return;
+        assert(
+          x === 0n || x === 1n,
+          `Field.assertBool(): expected ${x} to be 0 or 1`
+        );
+        return new Bool(x === 1n);
       }
       assertBoolean(this);
+      return Bool.Unsafe.fromField(this);
```

Comment:
I made `assertBool()` return the Bool because this is a common use case: assert that some witnessed field element is boolean and then use the bool